### DISCUSSION TITLE
Feature brainstorm

### DISCUSSION THREAD

**Discussion by SuzGupta** [mikaelacaron/Basic-Car-Maintenance#122] [Ideas]
Here's a variety of suggestions after building the current version and having a brief look and also looking through open issues.

NB: I think what we are tracking are "services" rather than "maintenances"... perhaps "tune-ups" (in which case this could be the alliterative "Tune-Up Tracker")

In Add Maintenance (which I would retitle Add Service) view:

- Add Type as a drop-down list with Other as text field. Suggested types (this needs more thinking through since often you do multiple of these during a single service): Oil Change, Tire Rotation, Tire Balancing, Brake Inspection & Replacement, Air Filter Replacement, Fluid Checks & Top-offs, Battery Inspection & Replacement, Belt & Hose Inspection, Spark Plug Replacement, Wheel Alignment, Transmission Service, Cooling System Service, Fuel System Cleaning, State Safety Inspections & Emissions Testing, Wiper Blade Replacement, Light Checks & Replacement, Suspension & Steering Components Inspection
- Add Location where service performed, with Maps integration. Maybe this should then become a dropdown or some other way to easily choose the same location again.
- Choose Vehicle and prompt user to add vehicle if none available
- Optionally add a picture (so, camera and camera roll access) to a service.
- Add Remind me in… (show date picker) to send a notification when this service is again needed. Would be great to have some intelligence built in here based on type of service.

In Add Vehicle view:

- [x] Add year
- [x] Add color
- Add a picture of vehicle (camera, camera roll)
- [x] Add VIN?

Make individual service entries and also all records / all records within a date range shareable using share sheet.

- [x] Allow searching across all entries.

Allow adding an owner to app so partners can add services.

Offer additional color themes, to go along with icons, maybe ability to make the whole app look like your car’s colors?

**Comment by SuzGupta** [comment]
Would also be handy to see the next 1-3 suggested services per vehicle wherever the vehicle list is displayed (currently in Settings, but feels like it should be its own tab)

**Comment by mikaelacaron** [comment]
> I think what we are tracking are "services" rather than "maintenances"..
While I agree with this, I think it's just semantics and kinda all falls under the same umbrella, a "service" is some "event" that has happened to your vehicle

Love the other ideas! I'm thinking of how to write some of these issues, where they won't collide too much with other issues being worked on to (somewhat) prevent some merge conflicts

**Comment by SuzGupta** [comment]
Let's add TipKit to this app somewhere too, probably once the app is more built out and some guidance is needed.

**Comment by SuzGupta** [comment]
If we were to do suggestions as to when a next service is due, there are APIs to help with this. ChatGPT says: "companies like CarMD, Edmunds, or Autodata offer APIs that aggregate service schedules from various manufacturers."
